controllers/datadogmonitor/monitor.go

```diff
@@ -0,0 +1,113 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2021 Datadog, Inc.
+
+package datadogmonitor
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sort"
+
+	datadogapiclientv1 "github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	datadoghqv1alpha1 "github.com/DataDog/datadog-operator/api/v1alpha1"
+)
+
+func buildMonitor(dm *datadoghqv1alpha1.DatadogMonitor) (*datadogapiclientv1.Monitor, *datadogapiclientv1.MonitorUpdateRequest) {
+	monitorType := datadogapiclientv1.MonitorType(string(dm.Spec.Type))
+	query := dm.Spec.Query
+	name := dm.Spec.Name
+	msg := dm.Spec.Message
+
+	// TODO
+	o := datadogapiclientv1.MonitorOptions{}
+
+	m := datadogapiclientv1.NewMonitor()
+	m.SetType(monitorType)
+	m.SetQuery(query)
+	m.SetName(name)
+	m.SetMessage(msg)
+	m.SetOptions(o)
```

Comment:
maybe it is overkill but I like to scope things

```suggestion
	m := datadogapiclientv1.NewMonitor()
	{
	    m.SetType(monitorType)
	    m.SetQuery(query)
	    m.SetName(name)
	    m.SetMessage(msg)
	    m.SetOptions(o)
	}
```

I looked if it was able to chain `Set()` function it is not possible :(